google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -1355,7 +1355,10 @@ public com.google.api.services.bigquery.model.QueryResponse call() {
       // here, but this is left as future work.
       JobId jobId = JobId.fromPb(results.getJobReference());
       Job job = getJob(jobId, options);
-      return job.getQueryResults();
+      TableResult tableResult = job.getQueryResults();
+      // Return the JobID of the successful job
+      tableResult.setJobId(jobId);
```

Comment:
Is the explicit setJobId even needed here?  Won't getQueryResults set it automatically, as its already got the awareness and has the special logic for results vs empty resultsets, etc?

Reply:
Agreed. Resolved the redundancy.